### DISCUSSION TITLE
classes/class_navigationmeshsourcegeometrydata3d

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#296] [User-contributed notes]
# classes/class_navigationmeshsourcegeometrydata3d

Experimental: This class may be changed or removed in future versions. Inherits: Resource< RefCounted< Object Container for parsed source geometry data used in navigation mesh baking. Description: ...

https://docs.godotengine.org/en/stable/classes/class_navigationmeshsourcegeometrydata3d.html

**Comment by LikiaBili** [comment]
A quick tip that I just found out:
You can create this by creating a new container and parsing in level data,
i.e.
```gdscript
var navigation_mesh = NavigationMesh.new()
var source_geometry = NavigationMeshSourceGeometryData3D.new()

NavigationServer3D.parse_source_geometry_data(navigation_mesh, source_geometry, $MyTestRootNode)
```
